### PR TITLE
doc: several documentation improvements

### PR DESCRIPTION
**Pull request by jasnell**
bundled together to make landing them easier.

### CODE REVIEW

**Comment by jasnell** [comment]
@joyent/node-collaborators

File: doc/api/stream.markdown [hunk 1]
```diff
@@ -167,6 +167,33 @@ again when more data is available.
...
+data to be consumed. The callback passed to handle the `'readable'`
+event must be prepared to handle a `null` response from
+`readable.read([size])`. For instance, in the following example, `foo.txt`
+is an empty file:
+
+```javascript
+var fs = require('fs');
+var rr = fs.createReadStream('foo.txt');
+rr.on('readable', function() {
+  var chunk;
```
Comment (trevnorris) [inline doc/api/stream.markdown]: Is this variable only for demonstration? Not being used.

**Comment by trevnorris** [comment]
One question. Otherwise LGTM

File: doc/api/stream.markdown (hunk 1 above)
Comment (jasnell) [inline doc/api/stream.markdown]: Ah...right, just mentally zoned on that one ;-). I'll remove the variable really quick

File: doc/api/stream.markdown [hunk 2]
```diff
@@ -167,6 +167,32 @@ again when more data is available.
 The `readable` event is not emitted in the "flowing" mode with the
 sole exception of the last one, on end-of-stream.
 
+Note that the `'readable'` event indicates only that data *can* be
+read from the stream. It does not indicate whether there is actual
+data to be consumed. The callback passed to handle the `'readable'`
```
Comment (chrisdickinson) [inline doc/api/stream.markdown]: The first two sentences are pretty confusing. Maybe:

> The `'readable'` event indicates that the stream has new information: either new data is available or the end of the stream has been reached. In the former case, `.read()` will return that data. In the latter case, `.read()` will return `null`. For instance, in the following example, ...

Comment (jasnell) [inline doc/api/stream.markdown]: :+1: will make that change

Comment (jasnell) [inline doc/api/stream.markdown]: @chrisdickinson ... done... if it looks good to you I'll get it landed

**Comment by jasnell** [comment]
Landed in f91fa52, f7af915, 9faae7a, 423c433, daed421, 340e9f0 and 6036e4f

**Pull request closed without merging**